src/Raven.Server/Documents/TransactionCommands/JsonPatchCommand.cs

```diff
@@ -0,0 +1,373 @@
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException($"Unsupported command type '{command.Type}'.");
+                }
+
+                if (command.Type == CommandTypes.Test)
+                    continue;
+
+                using (document.Data)
+                {
+                    isModified = true;
+                    // ReadObject has to be called after every command so changes between each command are applied for the next operation's use.
+                    // For example: executing an Add at /a/b which would result in 'b' being created, then executing a Replace at /a/b
+                    // Without ReadObject after every command, TryGet will not find 'b' for the Replace operation.
+                    document.Data = context.ReadObject(document.Data, _id, BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+                }
+            }
+
+            DocumentsStorage.PutOperationResults? putResult = null;
+            if (isModified)
+                putResult  = context.DocumentDatabase.DocumentsStorage.Put(context, _id, document.ChangeVector, document.Data);
+
+            if (putResult == null)
+            {
+                _patchResult = new PatchResult
+                {
+                    Status = PatchStatus.NotModified,
+                };
+            }
+            else
+            {
+                _patchResult = new PatchResult
+                {
+                    Status = PatchStatus.Patched,
+                    ModifiedDocument = document.Data?.Clone(_externalContext),
+                    ChangeVector = putResult.Value.ChangeVector,
+                    Collection = putResult.Value.Collection.Name,
+                    LastModified = putResult.Value.LastModified
+                };
+            }
+
+            return 1;
+        }
+
+        public string HandleReply(DynamicJsonArray reply, HashSet<string> modifiedCollections, DocumentDatabase database)
+        {
+            if (_patchResult.ModifiedDocument != null)
+                database.HugeDocuments.AddIfDocIsHuge(_id, _patchResult.ModifiedDocument.Size);
+            
+            if (_patchResult.Collection != null)
+                modifiedCollections?.Add(_patchResult.Collection);
+
+            var patchReply = new DynamicJsonValue
+            {
+                [nameof(BatchRequestParser.CommandData.Id)] = _id,
+                [nameof(BatchRequestParser.CommandData.ChangeVector)] = _patchResult.ChangeVector,
+                [nameof(Constants.Documents.Metadata.LastModified)] = _patchResult.LastModified,
+                [nameof(BatchRequestParser.CommandData.Type)] = nameof(CommandType.JsonPatch),
+                [nameof(PatchStatus)] = _patchResult.Status,
+                [nameof(PatchResult.Debug)] = _patchResult.Debug //TODO: needed?
```

Comment:
You are mixing the `nameof` fields here, which is confusing. What is this being read into, use that